catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/stage/SceneTransitionWithSoundBrickStageTest.java

```diff
@@ -0,0 +1,167 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.stage;
+
+import android.media.MediaPlayer;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.common.SoundInfo;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.StartScript;
+import org.catrobat.catroid.content.bricks.PlaySoundBrick;
+import org.catrobat.catroid.content.bricks.SceneTransitionBrick;
+import org.catrobat.catroid.content.bricks.WaitBrick;
+import org.catrobat.catroid.io.ResourceImporter;
+import org.catrobat.catroid.io.SoundManager;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.stage.StageActivity;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.uiespresso.stage.utils.ScriptEvaluationGateBrick;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityTestRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.IOException;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+import static org.catrobat.catroid.common.Constants.SOUND_DIRECTORY_NAME;
+
+@RunWith(AndroidJUnit4.class)
+public class SceneTransitionWithSoundBrickStageTest {
+
+	private ScriptEvaluationGateBrick lastBrickFirstScript;
+	private ScriptEvaluationGateBrick lastBrickSecondScript;
+
+	private String firstSceneName;
+	private Script secondScript;
+
+	private static ProjectManager projectManager;
+	private File soundFile;
+
+	@Rule
+	public BaseActivityTestRule<StageActivity> baseActivityTestRule = new
+			BaseActivityTestRule<>(StageActivity.class, false,
+			false);
+
+	@BeforeClass
+	public static void setUpProjectManager() {
+		projectManager = ProjectManager.getInstance();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		createProject(getClass().getSimpleName());
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@After
+	public void tearDown() throws IOException {
+		TestUtils.deleteProjects(getClass().getSimpleName());
+		if (soundFile != null && soundFile.exists()) {
+			soundFile.delete();
+		}
+	}
+
+	@Test
+	public void testStopSoundOnSceneTransition() {
+		runProject(lastBrickSecondScript);
```

Comment:
```suggestion
		runProject(lastBrickFirstScript);
```

PLEASE CHECK: Shouldn't this also start with the first scene here? Otherwise the sound never starts, right?

Reply:
In runProject the whole Project is run, both scripts get executed - the parameter is only used to wait until this Script is fully evaluated.

Reply:
Thank you for explaining!